global_schema.ddl

```diff
@@ -302,16 +511,104 @@ CREATE TABLE IF NOT EXISTS migration_minion_sync (
 );
 
 /*
- * Global Settings
+ * Metrics
  */
-CREATE TABLE IF NOT EXISTS global_setting (
+CREATE TABLE IF NOT EXISTS metrics_batch (
+	uuid			TEXT PRIMARY KEY,
+	model_uuid		TEXT NOT NULL,
+	unit			TEXT,
+	charm_url		TEXT,
+	sent			BOOLEAN,
+	delete_time		TIMESTAMP,
+	created			TIMESTAMP,
+	credentials		TEXT,
+	sla_credential	TEXT,
+	CONSTRAINT		fk_metrics_batch_model
+		FOREIGN KEY	(model_uuid)
+		REFERENCES	model(uuid)
+);
+
+CREATE TABLE IF NOT EXISTS metric (
+	uuid				TEXT PRIMARY KEY,
+	metrics_batch_uuid	TEXT NOT NULL,
+	key					TEXT,
+	value				TEXT,
+	time				TIMESTAMP,
+	CONSTRAINT			fk_metric_metric_batch
+		FOREIGN KEY		(metrics_batch_uuid)
+		REFERENCES		metrics_batch(uuid)
+);
+
+CREATE TABLE IF NOT EXISTS metric_label (
 	uuid			TEXT PRIMARY KEY,
-	name			TEXT NOT NULL,
-	value			TEXT
+	metric_uuid		TEXT NOT NULL,
+	key				TEXT,
+	value			TEXT,
+	CONSTRAINT		fk_metric_label_metric
+		FOREIGN KEY	(metric_uuid)
+		REFERENCES	metric(uuid)
+);
+
+CREATE TABLE IF NOT EXISTS metric_manager (
+	uuid					TEXT PRIMARY KEY,
+	last_successful_send	TIMESTAMP,
+	consecutive_errors		INT,
+	grace_period			INT
+);
+
+/*
+ * Leases
+ */
+CREATE TABLE IF NOT EXISTS lease_type (
+	id		INT PRIMARY KEY,
+	type	TEXT
+);
+
+INSERT INTO lease_type VALUES
+(
+	0,
+	"controller"
+), (
+	1,
+	"model"
+), (
+	2,
+	"application"
+);
+
+CREATE TABLE IF NOT EXISTS lease (
+	uuid			TEXT PRIMARY KEY,
+	lease_type_id	INT NOT NULL,
+	name			TEXT,
+	holder			TEXT,
+	start			TIMESTAMP,
+	duration		INT,
+	pinned			BOOLEAN,
+	CONSTRAINT		fk_lease_lease_type
+		FOREIGN KEY	(lease_type_id)
+		REFERENCES	lease_type(id)
+);
+
+/*
+ * Settings
+ */
+CREATE TABLE IF NOT EXISTS settings_batch (
```

Comment:
Let's name these `settings_group` and `settings_value`, and substitute "batch" with "group".

Reply:
Fixed!